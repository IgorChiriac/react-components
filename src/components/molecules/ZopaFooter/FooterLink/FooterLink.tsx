import styled from 'styled-components';
import * as colors from '../colors';
import Link from '../../../atoms/Link/Link';

export interface IFooterLink extends React.HTMLAttributes<HTMLAnchorElement> {
  size?: 'default' | 'small';
}

const FooterLink = styled(Link)`
  color: ${colors.neutral200};
  transition: color 0.2s ease;

  &:hover {
    color: ${colors.neutral400};
  }
`;

FooterLink.defaultProps = {
  size: 'default',
  weight: 'semibold',
};

export default FooterLink;
