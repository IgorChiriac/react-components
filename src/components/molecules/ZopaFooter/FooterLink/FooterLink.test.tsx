import { axe } from 'jest-axe';
import React from 'react';
import { render } from '@testing-library/react';
import FooterLink from './FooterLink';

describe('<FooterLink />', () => {
  it('renders the component with default props', async () => {
    const { container } = render(<FooterLink />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders the component with a small size', async () => {
    const { container } = render(<FooterLink size="small" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders the component with no a11y violations', async () => {
    const { container } = render(<FooterLink />);
    const results = await axe(container.innerHTML);
    expect(results).toHaveNoViolations();
  });
});
