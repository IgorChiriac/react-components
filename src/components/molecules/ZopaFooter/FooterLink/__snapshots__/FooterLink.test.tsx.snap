// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FooterLink /> renders the component with a small size 1`] = `
.c0 {
  font-size: inherit;
  font-family: "Open Sans",-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-weight: 600;
  line-height: 1.6;
  color: #4A3EDE;
  cursor: pointer;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c0:hover {
  opacity: 0.88;
}

.c0:active {
  opacity: 0.72;
}

.c1 {
  color: #A9ACBA;
  -webkit-transition: color 0.2s ease;
  transition: color 0.2s ease;
}

.c1:hover {
  color: #6F748B;
}

<a
  as="a"
  class="c0 c1"
  color="#4A3EDE"
/>
`;

exports[`<FooterLink /> renders the component with default props 1`] = `
.c0 {
  font-size: inherit;
  font-family: "Open Sans",-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-weight: 600;
  line-height: 1.6;
  color: #4A3EDE;
  cursor: pointer;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c0:hover {
  opacity: 0.88;
}

.c0:active {
  opacity: 0.72;
}

.c1 {
  color: #A9ACBA;
  -webkit-transition: color 0.2s ease;
  transition: color 0.2s ease;
}

.c1:hover {
  color: #6F748B;
}

<a
  as="a"
  class="c0 c1"
  color="#4A3EDE"
/>
`;
